### PR TITLE
h3: fixing a bug with prepopulated_entries and testing

### PR DESCRIPTION
**Pull request by alyssawilk**
alt-svc-cache prepopulated entries pre-populated with a non-empty hostname, a feature not yet supported by the connection grid.  This fixes and "tests"

Risk Level: low (unused feature given no one complained about brokenness)
Testing: yes
Docs Changes: n/a
Release Notes:

### CODE REVIEW

File: mobile/test/common/integration/client_integration_test.cc
Comment (RyanTheOptimist) [inline mobile/test/common/integration/client_integration_test.cc:252]: @danzh2010 suggested that if we use the default verifier then we can disable the validation check here:

https://source.corp.google.com/piper///depot/google3/third_party/envoy/src/api/envoy/extensions/transport_sockets/tls/v3/common.proto;l=524;rcl=572257956

Is that plausible?

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks for the fix!

File: source/common/http/http_server_properties_cache_manager_impl.cc
Comment (RyanTheOptimist) [reply source/common/http/http_server_properties_cache_manager_impl.cc:64]: We don't support it yet, but Alt-Svc allow https requests for www.example.org to be instead sent to alternate.example.com. We sometimes refer to this as "remote alt-svc".

File: mobile/test/common/integration/client_integration_test.cc
Comment (alyssawilk) [reply mobile/test/common/integration/client_integration_test.cc:256]: unfortunately not that simple.  even if we change the server to allow it, we end up with the mobile client rejecting the cert because of san mismatch.  when I comment that out we fail somewhere else :-(

Comment (RyanTheOptimist) [reply mobile/test/common/integration/client_integration_test.cc:256]: Blergh. Why is nothing simple?! Makes sense. Hopefully we'll be able to fix this in subsequent changes.

**Review by RyanTheOptimist — APPROVED** [review]

**Comment by alyssawilk** [comment]
/wait on CI - I think we need some release_envoy_data in here

**Comment by RyanTheOptimist** [comment]
I think this needs a merge once #30161 lands.

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**
